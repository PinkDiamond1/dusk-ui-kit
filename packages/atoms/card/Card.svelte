<script>
  import { variants } from "@dusk-network/helpers";
  import "./styles.css";

  export let variant = variants.ATOM.CARD.DEFAULT;
  export let href = null;
  export let disabled = false;
</script>

{#if href}
  <a href="{href}">
    <div
      class="{$$props.class || ''} duk-card"
      class:duk-card--brand="{variant === variants.ATOM.CARD.BRAND}"
      class:duk-card--cta="{variant === variants.ATOM.CARD.CTA}"
      class:duk-card--info="{variant === variants.ATOM.CARD.INFO}"
      class:duk-card--success="{variant === variants.ATOM.CARD.SUCCESS}"
      class:duk-card--warning="{variant === variants.ATOM.CARD.WARNING}"
      class:duk-card--danger="{variant === variants.ATOM.CARD.DANGER}"
      class:duk-card--disabled="{disabled}"
      on:*
    >
      <slot />
    </div>
  </a>
{:else}
  <div
    class="{$$props.class || ''} duk-card"
    class:duk-card--brand="{variant === variants.ATOM.CARD.BRAND}"
    class:duk-card--cta="{variant === variants.ATOM.CARD.CTA}"
    class:duk-card--info="{variant === variants.ATOM.CARD.INFO}"
    class:duk-card--success="{variant === variants.ATOM.CARD.SUCCESS}"
    class:duk-card--warning="{variant === variants.ATOM.CARD.WARNING}"
    class:duk-card--danger="{variant === variants.ATOM.CARD.DANGER}"
    class:duk-card--disabled="{disabled}"
    on:*
  >
    <slot />
  </div>
{/if}
