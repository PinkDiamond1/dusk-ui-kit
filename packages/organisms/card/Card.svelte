<script>
  import { setContext } from "svelte";
  import contexts from "@dusk-network/helpers/contexts.js";
  import variants from "@dusk-network/helpers/variants.js";
  import states from "@dusk-network/helpers/states.js";
  import "./styles.css";

  export let variant = variants.CARD.DEFAULT;
  export let state = states.CARD.DEFAULT;
  export let href = null;
  export let disabled = false;

  setContext("DUK:accordion:context", contexts.CARD.ACCORDION);
  setContext("DUK:rich-text:context", contexts.RICH_TEXT.CARD);
  setContext("DUK:heading:context", contexts.HEADING.CARD);
</script>

{#if href}
  <a href="{href}">
    <div
      class="{$$props.class || ''} duk-card"
      class:duk-card--brand="{variant === variants.CARD.BRAND}"
      class:duk-card--cta="{variant === variants.CARD.CTA}"
      class:duk-card--info="{variant === variants.CARD.INFO}"
      class:duk-card--success="{variant === variants.CARD.SUCCESS}"
      class:duk-card--warning="{variant === variants.CARD.WARNING}"
      class:duk-card--danger="{variant === variants.CARD.DANGER}"
      class:duk-card--passed="{state === states.CARD.PASSED}"
      class:duk-card--error="{state === states.CARD.ERROR}"
      class:duk-card--failed="{state === states.CARD.FAILED}"
      class:duk-card--disabled="{disabled}"
      on:*
    >
      <slot />
    </div>
  </a>
{:else}
  <div
    class="{$$props.class || ''} duk-card"
    class:duk-card--brand="{variant === variants.CARD.BRAND}"
    class:duk-card--cta="{variant === variants.CARD.CTA}"
    class:duk-card--info="{variant === variants.CARD.INFO}"
    class:duk-card--success="{variant === variants.CARD.SUCCESS}"
    class:duk-card--warning="{variant === variants.CARD.WARNING}"
    class:duk-card--danger="{variant === variants.CARD.DANGER}"
    class:duk-card--passed="{state === states.CARD.PASSED}"
    class:duk-card--error="{state === states.CARD.ERROR}"
    class:duk-card--failed="{state === states.CARD.FAILED}"
    class:duk-card--disabled="{disabled}"
    on:*
  >
    <slot />
  </div>
{/if}
