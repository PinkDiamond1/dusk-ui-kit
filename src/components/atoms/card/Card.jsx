/* @flow */
/** @jsx jsx */
import { css, jsx } from '@emotion/core'
import * as React from 'react'
import BootstrapCard from 'react-bootstrap/Card'

export const CardVariants = {
  PRIMARY: 'primary',
  SECONDARY: 'secondary',
  INFO: 'info',
  SUCCESS: 'success',
  WARNING: 'warning',
  DANGER: 'danger',
}

type Props = {
  className: string,
  title: string,
  variant: 'primary' | 'secondary' | 'info' | 'success' | 'warning' | 'danger',
  children: React.Node,
}

const Card = (props: Props): React.Element<*> => {
  const { className, title, variant, children } = props
  let bg
  let border
  let text
  let headerClass = ''
  switch (variant) {
    case 'primary':
      bg = 'primary'
      border = 'primary'
      text = 'white'
      headerClass = 'text-white'
      break
    case 'secondary':
      bg = 'secondary'
      border = 'secondary'
      text = 'white'
      headerClass = 'text-white'
      break
    case 'info':
      bg = 'info'
      border = 'info'
      text = 'white'
      headerClass = 'text-white'
      break
    case 'success':
      bg = 'success'
      border = 'success'
      text = 'white'
      headerClass = 'text-white'
      break
    case 'warning':
      bg = 'warning'
      border = 'warning'
      text = 'white'
      headerClass = 'text-white'
      break
    case 'danger':
      bg = 'danger'
      border = 'danger'
      text = 'white'
      headerClass = 'text-white'
      break
    case 'dark':
      bg = 'dark'
      border = 'dark'
      text = 'light'
      headerClass = 'text-light'
      break
    case 'light':
      bg = 'light'
      border = 'dark'
      text = 'dark'
      headerClass = 'text-dark'
      break
    default:
      bg = ''
      border = ''
      text = ''
      headerClass = ''
  }

  const updatedClassName = `__duk-card ${className}`

  return (
    <BootstrapCard
      className={updatedClassName}
      bg={bg}
      border={border}
      text={text}
      css={css`
        box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
      `}
    >
      <BootstrapCard.Body>
        {title !== '' &&
          <BootstrapCard.Title
            className={headerClass}
            css={css`
              {
                font-weight: 500;
                font-size: 1rem;
              }
            `}
          >
            {title}
          </BootstrapCard.Title>
        }
        {children}
      </BootstrapCard.Body>
    </BootstrapCard>
  )
}

Card.defaultProps = {
  className: '',
  title: '',
  variant: '',
}

export default Card
