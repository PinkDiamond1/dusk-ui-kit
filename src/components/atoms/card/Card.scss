.__duk-card {
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);

  .card-title {
    font-weight: 500;
    font-size: 1rem;
  }
}
