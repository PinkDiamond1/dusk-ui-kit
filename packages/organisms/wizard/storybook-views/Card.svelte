<script>
  import Default from "./Default.svelte";
  import Card, { Content } from "@dusk-network/card";
</script>

<Card>
  <Content>
    <Default />
  </Content>
</Card>
