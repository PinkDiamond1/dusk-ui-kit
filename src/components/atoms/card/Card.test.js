import React from 'react'
import renderer from 'react-test-renderer'

import Card from './Card'

it('test to see if the Card renders correctly', () => {
  const tree = renderer.create(<Card />).toJSON()
  expect(tree).toMatchSnapshot()
})
