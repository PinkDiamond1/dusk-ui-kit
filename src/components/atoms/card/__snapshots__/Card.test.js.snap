// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the Card renders correctly 1`] = `
<div
  className="__duk-card  card"
>
  <div
    className="card-body"
  />
</div>
`;
